### PR TITLE
Disable compression for state db

### PR DESCRIPTION
**Pull request by asdacap**
- By default rocksdb turn on snappy compression.
- This PR add a disable compression option and enable it by default for state db.
- The effect so tiny its impressive (that snappy is this transparent), I'm not even sure if its noise, so.. toss a coin or something.

Metrics when running verify trie without memory budget as standard visitor always cache miss, making it a pretty good stress test. Percentile is nanosec per get. The last column is total cpu time per node in microsec. 
| | Db Size | 25p | 50p | 75p | 95p | Microsec per node | 
|-|-|-|-|-|-|-|
| With compression | 168GB | 27.2 | 44.0 | 51.8 | 61.1 | 38.4 | 
| No compression | 172 GB | 29.1 | 44.0 | 49.4 | 59.7 | 37.1 | 

Before -> Full prune -> After
Screenshot from 2023-05-09 21-35-40

## Changes

- Add option to disable compression with state db.
- Disable compression for state db by default.

#### What types of changes does your code introduce?

- [X] Optimization

#### Requires testing

- [X] Yes
- [ ] No

#### If yes, did you write tests?

- [ ] Yes
- [X] No

#### Notes on testing

- SSD used is Optane 905p with XFS.

### CODE REVIEW

File: src/Nethermind/Nethermind.Db.Rocks/Config/DbConfig.cs [hunk 1]
```diff
@@ -154,6 +155,7 @@ public class DbConfig : IDbConfig
     public int? StateDbBlockSize { get; set; } = 4 * 1024;
     public bool? StateDbUseDirectReads { get; set; } = false;
     public bool? StateDbUseDirectIoForFlushAndCompactions { get; set; } = false;
+    public bool? StateDbDisableCompression { get; set; } = true;
```
Comment (benaadams) [inline src/Nethermind/Nethermind.Db.Rocks/Config/DbConfig.cs:158]: This isn't used?

**Review by benaadams — CHANGES_REQUESTED** [review]
Only uses the `DisableCompression` setting; not the more specific `StateDbDisableCompression`?

File: src/Nethermind/Nethermind.Db.Rocks/Config/DbConfig.cs (hunk 1 above)
Comment (asdacap) [reply src/Nethermind/Nethermind.Db.Rocks/Config/DbConfig.cs:158]: No this is used. The `PerTableConfig` class check the field `<dbname>DisableCompression` via reflection. I've checked manually via custom log.

**Review by benaadams — APPROVED** [review]

**Comment by LukaszRozmej** [comment]
Is this backwards compatible? When we upgrade node that was synced with compression to one that now is by default without compression? If not than I think it is not worth it.

**Comment by asdacap** [comment]
Its forward and backward compatible. You can have partly compressed db.

**Pull request merged**